Makefile

```diff
@@ -113,6 +113,74 @@ distclean:
 
 check: test
 
+# Remove files generated by running coverage, put the non-instrumented lib back 
+# in place 
+coverage-clean:
+	if [ -d lib_ ]; then rm -rf lib; mv lib_ lib; fi
+	-rm -rf gcovr testing
+	-rm -rf out/$(BUILDTYPE)/.coverage
+	-rm -rf .cov_tmp coverage
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/*.gcda
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/tracing/*.gcda
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/*.gcno
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/tracing*.gcno
+
+# Build and test with code coverage reporting.  Leave the lib directory 
+# instrumented for any additional runs the user may want to make.
+# For C++ coverage reporting, this needs to be run in conjunction with configure
+#  --coverage.  html coverage reports will be created under coverage/
+
+coverage: coverage-build coverage-test
+
+coverage-build: all
+	mkdir -p node_modules
+	if [ ! -d node_modules/istanbul-merge ]; then \
+		$(NODE) ./deps/npm install istanbul-merge; fi
+	if [ ! -d node_modules/nyc ]; then $(NODE) ./deps/npm install nyc; fi
+	if [ ! -d gcovr ]; then git clone --depth=1 \
+		--single-branch git://github.com/gcovr/gcovr.git; fi
+	if [ ! -d testing ]; then git clone --depth=1 \
+		--single-branch https://github.com/nodejs/testing.git; fi
+	if [ ! -f gcovr/scripts/gcovr.orig ]; then \
+		(cd gcovr && patch -N -p1 < \
+		"$(CURDIR)/testing/coverage/gcovr-patches.diff"); fi
+	if [ -d lib_ ]; then rm -rf lib; mv lib_ lib; fi
+	mv lib lib_
+	$(NODE) ./node_modules/.bin/nyc instrument lib_/ lib/ 
+	$(MAKE)
+
+coverage-test:
+	$(MAKE) build-addons
+	$(MAKE) cctest
+	-rm -rf out/$(BUILDTYPE)/.coverage
+	-rm -rf .cov_tmp
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/*.gcda
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/tracing/*.gcda
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/*.gcno
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/tracing*.gcno
+	-$(PYTHON) tools/test.py --mode=release -J addons doctool inspector \
+		known_issues pseudo-tty parallel sequential
```

Comment:
`-J` should use the `$JOBS` environment variable or the number of CPUs, it definitely shouldn't be spawning as many processes as possible. 

Source here: https://github.com/nodejs/node/blob/master/tools/test.py#L1429-L1433